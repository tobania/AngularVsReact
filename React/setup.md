# React Setup

## Prerequirements
- nodejs
- Backend runs in the background and is accessable

# Setup
Navigate to the frontend root (src) and execute the following commands:
npm install
npm start
