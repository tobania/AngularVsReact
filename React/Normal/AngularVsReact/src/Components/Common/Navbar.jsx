import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class Navbar extends PureComponent {
  render() {
    const { showNewTodo } = this.props;

    return (
      <nav className="navbar navbar-dark bg-dark">
        <span className="navbar-brand mb-0 h1">Todo app</span>
        <button type="button" className="btn btn-success" onClick={showNewTodo}>Add todo list</button>
      </nav>
    );
  }
}

Navbar.propTypes = {
  showNewTodo: PropTypes.func.isRequired,
};

Navbar.defaultProps = {
};

export default Navbar;
