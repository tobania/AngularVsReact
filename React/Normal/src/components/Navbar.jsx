import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Navbar extends Component {
  render() {
    const { showNewTodo } = this.props;

    return (
      <nav className="navbar navbar-dark bg-dark">
        <span className="navbar-brand mb-0 h1">Todo app</span>
        <button type="button" className="btn btn-success" onClick={showNewTodo}>Add todo list</button>
      </nav>
    );
  }
}

Navbar.propTypes = {
  showNewTodo: PropTypes.func.isRequired,
};

Navbar.defaultProps = {
};

export default Navbar;
