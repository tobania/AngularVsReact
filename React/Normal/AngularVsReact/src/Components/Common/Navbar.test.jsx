import React from 'react';
import renderer from 'react-test-renderer';
import Navbar from './Navbar';

describe('Navbar', () => {
  it('gets loaded correctly', () => {
    const component = renderer.create(
      <Navbar showNewTodo={jest.fn} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
