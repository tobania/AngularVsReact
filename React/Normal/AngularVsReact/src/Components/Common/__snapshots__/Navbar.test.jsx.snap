// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar gets loaded correctly 1`] = `
<nav
  className="navbar navbar-dark bg-dark"
>
  <span
    className="navbar-brand mb-0 h1"
  >
    Todo app
  </span>
  <button
    className="btn btn-success"
    onClick={[Function]}
    type="button"
  >
    Add todo list
  </button>
</nav>
`;
